perun-core/src/main/java/cz/metacentrum/perun/core/impl/GroupsManagerImpl.java

```diff
@@ -1027,4 +1027,33 @@ public MemberGroupStatus getTotalMemberGroupStatus(PerunSession session, Member
 			throw new InternalErrorException(e);
 		}
 	}
+
+	@Override
+	public List<Group> getGroupsForAutoRegistration(PerunSession sess, Vo vo) {
+		try {
+			return jdbc.query("select " + groupMappingSelectQuery + " from groups where vo_id=? and id IN (select group_id from groups_to_register)", GROUP_MAPPER, vo.getId());
+		} catch (RuntimeException ex) {
+			throw new InternalErrorException(ex);
+		}
+	}
+
+	@Override
+	public void deleteGroupFromAutoRegistration(PerunSession sess, Group group) throws GroupAlreadyRemovedException {
+		try {
+			int rowAffected = jdbc.update("delete from groups_to_register where group_id=?", group.getId());
+
+			if(rowAffected == 0) throw new GroupAlreadyRemovedException("Group: " + group);
```

Comment:
This exception is meant for a different purpose. I think it is not necessary to throw an exception when this happens. You can just silently skip it.

Reply:
Ok, removed.